packages/bunnymen-core/src/node.ts

```diff
@@ -1,246 +1,294 @@
-          data += decoder.decode(chunk, {
-            stream: true
-          })
-        }
+    this._currentCid = cid
+    this.emit('downloadedData', data)
+    return data
+  }
 
-        this._currentCid = cid
-        this.emit('downloadedData', data)
-        return data
-    }
+  getPeers() {
+    return this._peers
+  }
 
-    getPeers(){
-        return this._peers
-    }
+  isLeader() {
+    return this._currentLeader == this._peerId.toString()
+  }
 
-    isLeader(){
-        return this._currentLeader == this._peerId.toString()
-    }
+  private selectLeader() {
+    var peers = this._peers
+    // add local peerId to peer list
+    peers.push(this._peerId)
+    // create a list of objects { peerId, hash }
+    // where we hash each peer with the currentCid
+    const peerHashList = peers.map((peer) => {
+      var preimage = peer + this._currentCid
+      return { peerId: peer, hash: String.fromCharCode(...sha3_256(preimage)) }
+    })
+    // sort the list alphanumerically
+    const peerHashListSorted = peerHashList.sort((a, b) => {
+      return a.hash.localeCompare(b.hash, undefined, {
+        numeric: true,
+        sensitivity: 'base',
+      })
+    })
+    // select the first peerId in the list as the new leader
+    this._currentLeader = peerHashListSorted[0].peerId
+    this.emit('selectedLeader', this._currentLeader)
+  }
 
-    private selectLeader(){
-        var peers = this._peers
-        // add local peerId to peer list
-        peers.push(this._peerId)
-        // create a list of objects { peerId, hash }
-        // where we hash each peer with the currentCid
-        const peerHashList = peers.map(peer => {
-            var preimage = peer+this._currentCid
-            return {peerId: peer, hash: String.fromCharCode(...sha3_256(preimage))}
-        })
-        // sort the list alphanumerically
-        const peerHashListSorted = peerHashList.sort((a,b)=>{
-            return a.hash.localeCompare(b.hash,undefined, {
-                numeric: true,
-                sensitivity: 'base'
-            })
-        })
-        // select the first peerId in the list as the new leader
-        this._currentLeader = peerHashListSorted[0].peerId
-        this.emit('selectedLeader', this._currentLeader)
-    }
+  private async checkForNewPeers() {
+    const node: IPFS.IPFS = await this._node
+    var prevPeers: string[] = this._peers
+    this._peers = (await node.pubsub.peers(this._topic)).map(String)
 
-    private async checkForNewPeers(){
+    var peersLeft = prevPeers.filter(
+      (prevPeer: string) =>
+        prevPeer != this._peerId && !this._peers.includes(prevPeer)
+    )
+    peersLeft.forEach((peer) => this.emit('peerUnsubscribed', peer.toString()))
 
-        const node: IPFS.IPFS = await this._node
-        var prevPeers: string[] = this._peers;
-        this._peers = (await node.pubsub.peers(this._topic)).map(String);
+    var peersJoined = this._peers.filter(
```

Comment:
~~var~~ **const** 🙏

Reply:
ill change this in the next PR. this isn't new code though. i think formatters are fighting